<a href="https://colab.research.google.com/github/nhatminh-it/fruits_classification/blob/master/eval_cluster_color.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/drive


In [ ]:
import pandas as pd
from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVC
from sklearn.metrics import accuracy_score
import cv2

In [ ]:
cluster_features_1_train = pd.read_csv('/content/drive/My Drive/Fruit recognition/output_input/25_cluster_features_1_train.csv')
cluster_features_1_test = pd.read_csv('/content/drive/My Drive/Fruit recognition/output_input/25_cluster_features_1_test.csv')

cluster_features_2_train = pd.read_csv('/content/drive/My Drive/Fruit recognition/output_input/25_cluster_features_2_train.csv')
cluster_features_2_test = pd.read_csv('/content/drive/My Drive/Fruit recognition/output_input/25_cluster_features_2_test.csv')

In [ ]:
models = []
models.append(('rfc', RandomForestClassifier()))
models.append(('svc', SVC()))

In [ ]:
print(cluster_features_1_train.shape)
print(cluster_features_1_test.shape)
print(cluster_features_2_train.shape)
print(cluster_features_2_test.shape)

(1870, 51)
(630, 51)
(1870, 7)
(630, 7)


# Cluster 1

In [ ]:
Y_train = cluster_features_1_train['name']
X_train = cluster_features_1_train.drop(['name'], axis=1)
Y_test = cluster_features_1_test['name']
X_test = cluster_features_1_test.drop(['name'], axis=1)

print(X_train.shape)
print(X_test.shape)
print(Y_train.shape)
print(Y_test.shape)

(1870, 50)
(630, 50)
(1870,)
(630,)


In [ ]:
for name, model in models:
  model.fit(X_train, Y_train)
  print(name)
  print('train_acc', accuracy_score(Y_train, model.predict(X_train)))
  print('test_acc', accuracy_score(Y_test, model.predict(X_test)))

rfc
train_acc 1.0
test_acc 0.8365079365079365
svc
train_acc 0.9096256684491979
test_acc 0.8269841269841269


# Cluster 2

In [ ]:
Y_train = cluster_features_2_train['name']
X_train = cluster_features_2_train.drop(['name'], axis=1)
Y_test = cluster_features_2_test['name']
X_test = cluster_features_2_test.drop(['name'], axis=1)

print(X_train.shape)
print(X_test.shape)
print(Y_train.shape)
print(Y_test.shape)

(1870, 6)
(630, 6)
(1870,)
(630,)


In [ ]:
for name, model in models:
  model.fit(X_train, Y_train)
  print(name)
  print('train_acc', accuracy_score(Y_train, model.predict(X_train)))
  print('test_acc', accuracy_score(Y_test, model.predict(X_test)))

rfc
train_acc 1.0
test_acc 0.9142857142857143
svc
train_acc 0.8978609625668449
test_acc 0.9047619047619048


# Tổng kết
* Cluster 1 (50 columns)

  RandomForestClassfier
    - Train accuraccy: 1.0
    - Test accuraccy: 0.8365079365079365

  SVC
    - Train accuraccy: 0.9096256684491979
    - Test accuraccy: 0.8269841269841269

* Cluster 2 (6 columns)

  RandomForestClassfier
    - Train accuraccy: 1.0
    - Test accuraccy: 0.9142857142857143

  SVC
    - Train accuraccy: 0.8978609625668449
    - Test accuraccy: 0.9047619047619048

**Đánh giá tổng quan Cluster 1 cho ra kết quả tốt hơn.**
